pkg/manifests/external_dns.go

```diff
@@ -54,41 +87,50 @@ func (p Provider) String() string {
 	}
 }
 
-func (p Provider) ResourceName() string {
-	switch p {
+// ExternalDnsConfig defines configuration options for required resources for external dns
+type ExternalDnsConfig struct {
+	TenantId, Subscription, ResourceGroup,
+	ClientId, ServiceAccountName, Namespace string
+	IdentityType       IdentityType
+	ResourceType       ResourceType
+	Provider           Provider
+	DnsZoneResourceIDs []string
+}
+
+func externalDNSResourceName(e *ExternalDnsConfig) string {
```

Comment:
why not make this a method?

Reply:
good call - this can be made into a method I'll adjust accordingly